lib/knife/changelog/policyfile.rb

```diff
@@ -1,69 +1,158 @@
 # frozen_string_literal: true
```

Comment:
The code is very clean and contract of methods very clear, this is really great and it can part of counterbalance the following! The following is optional: as discussed together to clarify the contract of the interface, I would have make almost all methods private. Actually only one method needs to be in the public contract of this class. If you did this for testability (what I can understand) I would have spreaded the concerns into several classes that have their own public contracts and which are therefore testable and composable. That way you both solve testability and 1 class = 1 responsibility which will ease the identification of impacts of future fixes (https://en.wikipedia.org/wiki/Single_responsibility_principle)